sdk/samples/iot/vxworks/azureProvisioningClientSample.c

```diff
@@ -0,0 +1,996 @@
+ *
+ * mqttPublish - publish a message to Azure IoT Hub
+ *
+ * This routine publishes a messsage to Azure IoT Hub Provisioning Service.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: true if the publication is successful, otherwise false
+ *
+ * \NOMANUAL
+ */
+
+static bool mqttPublish(
+    struct mosquitto* mqttClient, /* MQTT client object */
+    int* msgId, /* message ID */
+    char* topic, /* topic to publish */
+    int payloadLen, /* payload length */
+    void* payload /* payload data */
+)
+{
+  int res;
+  res = mosquitto_publish(mqttClient, msgId, topic, payloadLen, payload, 1, true);
+  if (res == MOSQ_ERR_SUCCESS)
+  {
+    printf_s("mosquitto_publish: %d OK!\n", *msgId);
+  }
+  else
+  {
+    printf_s("mosquitto_publish ERROR: %s\n", mosquitto_strerror(res));
+    return false;
+  }
+  return true;
+}
+
+/*******************************************************************************
+ *
+ * onConnect - connect callback of a MQTT client
+ *
+ * This routine is called when a MQTT client is connected to the
+ * Azure IoT Hub Provisioning Service.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void onConnect(
+    struct mosquitto* mqttClient, /* MQTT client object */
+    void* obj, /* user data provided in mosquitto_new */
+    int result /* result of the connection */
+)
+{
+  if (result == MOSQ_ERR_SUCCESS)
+  {
+    printf_s("connect OK!\n");
+  }
+  else
+  {
+    printf_s("connect ERROR: %s\n", mosquitto_strerror(result));
+  }
+}
+
+/*******************************************************************************
+ *
+ * onPublish - publish callback of a MQTT client
+ *
+ * This routine is called when a MQTT client finishes sending a message to the
+ * Azure IoT Hub Provisioning Service.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void onPublish(
+    struct mosquitto* mqttClient, /* MQTT client object */
+    void* obj, /* user data provided in mosquitto_new */
+    int msgId /* message ID */
+)
+{
+  printf_s("Message: %d is sent!\n", msgId);
+}
+
+/*******************************************************************************
+ *
+ * queryStatusSend - publish a query status message to a Provisioning Service
+ *
+ * This routine publishes a query status message to a Provisioning Service to
+ * check whether a provisioning operation is completed or not.
+ *
+ * ERRNO: N/A
+ *
+ * RETURNS: N/A
+ *
+ * \NOMANUAL
+ */
+
+static void queryStatusSend(
+    az_iot_provisioning_client* azureClient, /* provisioning client */
+    struct mosquitto* mqttClient, /* MQTT client object */
+    az_iot_provisioning_client_register_response* response /* response */
+)
+{
+  char queryTopic[256];
+  int msgId;
+
+  if (az_result_failed(az_iot_provisioning_client_query_status_get_publish_topic(
+          azureClient, response->operation_id, queryTopic, sizeof(queryTopic), NULL)))
+  {
+    printf_s("az_iot_provisioning_client_query_status_get_publish_topic "
```

Comment:
too verbose

Reply:
This is a failure check. az_iot_provisioning_client_query_status_get_publish_topic will fail in rare cases. so we usually can't see the error message.